weave-js/src/util/date.ts

```
@@ -0,0 +1,387 @@
+
+    switch (unit) {
+      case 'd':
+        const daysAgo = new Date(now);
+        daysAgo.setDate(daysAgo.getDate() - amount);
+        return daysAgo;
+      case 'w':
+        const weeksAgo = new Date(now);
+        weeksAgo.setDate(weeksAgo.getDate() - amount * 7);
+        return weeksAgo;
+      case 'm':
+        const monthsAgo = new Date(now);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)